cmake/modules/RootMacros.cmake

```diff
@@ -484,8 +484,17 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
                      DEPENDS ${_list_of_header_dependencies} ${_linkdef} ${ROOTCINTDEP} ${MODULE_LIB_DEPENDENCY})
   get_filename_component(dictname ${dictionary} NAME)
 
-  #---roottest compability
-  add_custom_target(${dictname} DEPENDS ${dictionary}.cxx ${pcm_name} ${rootmap_name} ${cpp_module_file})
+  if(ARG_NOTARGET)
+    if(TARGET ${ARG_MODULE})
+      target_sources(${ARG_MODULE} PRIVATE ${dictionary}.cxx)
+    else()
+      message(FATAL_ERROR
+        " When used with NOTARGET, the MODULE option must passed a name of an existing target.\n"
```

Comment:
```suggestion
        " When used with NOTARGET, the MODULE option must be passed a name of an existing target.\n"
```

Reply:
Thanks, I also noticed it and will submit a new round of changes with this fixed.